### PR TITLE
Adding saveMany()

### PR DESCRIPTION
**Pull request by burzum**
@lorenzo as discussed here is the method from the trait we use in our plugin.

### CODE REVIEW

File: src/Type.php [hunk 1]
```diff
@@ -436,6 +436,78 @@ public function exists($conditions)
...
+                    'Invalid items in the list. Found `%s` but expected `%s`',
+                    is_object($entity) ? get_class($entity) : gettype($entity),
+                    EntityInterface::class
+                ));
+            }
+
+            $this->dispatchEvent('Model.beforeSave', [
+                'entity' => $entity,
+                'options' => $options
+            ]);
```
Comment (markstory) [inline src/Type.php]: Shouldn't you check that this event wasn't stopped? The ORM will abort saves when `Model.beforeSave` is cancelled.

Comment (burzum) [inline src/Type.php]: @markstory @lorenzo worked with us, he gave us a few lessons in ES and asked me if I would put this method from our plugin into the core plugin. He said he's going to finish it. I might look at it if I find the time.

**Comment by burzum** [comment]
@markstory @lorenzo worked with us, he gave us a few lessons in ES and asked me if I would put this method from our plugin into the core plugin. He said he's going to finish it. I might look at it if I find the time.

**Comment by lorenzo** [comment]
I can work on the tests this week. @burzum can you address the comment from mark regarding stopping the event?

**Comment by lorenzo** [comment]
Will fix the phpunit issue separatedly

**Pull request merged**